elide-async/src/main/java/com/yahoo/elide/async/service/AsyncQueryThread.java

```diff
@@ -68,11 +74,20 @@ protected void processQuery() {
                 response = elide.get(getPath(queryObj.getQuery()), queryParams, user, apiVersion);
                 log.debug("JSONAPI_V1_0 getResponseCode: {}, JSONAPI_V1_0 getBody: {}",
                         response.getResponseCode(), response.getBody());
+
+                if (isValidJSON(response.getBody())) {
```

Comment:
isValidJson check can be done from inside the calculateRecordsJSON.

Reply:
Changed this. Check is done directly by the calcluateRecords method